Simplify Login popup

jessgusclark opened the pull request:
This simplifies the login modal and focuses the user on one thing, logging in.

* Removed the contract address
* Removed 'register' button
* Used correct string to display user message that they don't own the domain
* Informs the user if their wallet is connected to the wrong network using existing strings.

ilanolkies commented on src/app/auth/components/AuthModalComponent.js:
```diff
@@ -46,6 +45,13 @@ class AuthModalComponent extends Component {
+    } else if (network === 'invalid') {
+      body = (
+        <div>
+          <p><strong>{strings.network_mismatch}</strong></p>
```
Isn’t it better to display the same text we display in the new indicator?

jessgusclark replied:
Yes, I can change this.

That string is in the `indicator-light` branch which is pending the translations. I will wait for that branch to be merged, then update line 51 to use the invalid string and push the commit.

jessgusclark replied:
Resolved with commit [d19b6d6](https://github.com/rnsdomains/rns-manager-react/pull/124/commits/d19b6d67ce40a18ad3f9848a0cbec75bc19ca1b5)

ilanolkies commented:
Closes #33

jessgusclark commented:
@ilanolkies I added the functionality when hitting enter to submit the form. I also added a spinner as the authentication process sometimes takes a half a second. Finally, no metamask redirects to the setup page.

jessgusclark commented on src/app/auth/reducer.js:
```diff
@@ -3,6 +3,7 @@ import * as types from './types';
 const initialState = {
   showModal: false,
   hasMetamask: true,
+  walletUnlocked: false,
```
Needs to be set to false because in Firefox with Metamask `RECEIVE_ENABLE` is no returned when the wallet is locked.

ilanolkies commented on src/app/auth/components/AuthModalComponent.js:
```diff
@@ -46,41 +55,32 @@ class AuthModalComponent extends Component {
-              </Col>
-            </Row>
-          </Form.Group>
+        <Form onSubmit={this.handleFormSubmit}>
```
Great!
